src/models/callStack.js

```diff
@@ -20,21 +20,64 @@ module.exports = class CallStack {
 
   // Attempts to remove any sensitive data that may be found within
   sanitize(content) {
-    if (typeof content !== "object") {
-      return content;
-    }
 
+    const hideString = "*****";
     let outContent = {};
+    let type = typeof content;
+
+    // Since JavaScript `typeof` will assign an array as "object" as well as null
+    // we will extend this typeof check to add those as different types, to ease
+    // the complexity of the below switch statement
+    if (Array.isArray(content)) {
+      type = "array";
+    }
+    if (content === null) {
+      type = "null";
+    }
+
+    switch(type) {
+      case "object":
+        for (const key in content) {
+          // Match different possible keys that represent sensitive data
+          switch(key) {
+            case "token":
+              outContent[key] = hideString;
+              break;
+            default:
+              outContent[key] = this.sanitize(content[key]);
+              break;
+          }
+        }
+        break;
+      case "string":
+        // Match different strings of sensitive data
 
-    for (const key in content) {
-      switch (key) {
-        case "token":
-          outContent[key] = "*****";
-          break;
-        default:
-          outContent[key] = content[key];
-          break;
-      }
+        // https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github#githubs-token-formats
+        if (content.startsWith("gho_")) {
+          outContent = hideString;
+        } else if (content.startsWith("ghp_")) {
+          outContent = hideString;
+        } else if (content.startsWith("github_pat_")) {
+          outContent = hideString;
+        } else if (content.startsWith("ghu_")) {
+          outContent = hideString;
+        } else if (content.startsWith("ghs_")) {
+          outContent = hideString;
+        } else if (content.startsWith("ghr_")) {
```

Comment:
perhaps this would be nicer as a regex?

```js
outContent = content.replace(/\b(?:github_pat_|ghp_|gho_|ghu_|ghs_|ghr_)\W*?\b/, hideString);
```


^ that one has the benefit of hiding tokens in case the string contains a token, but also other things too

Reply:
Changing this into regex is a great idea to help find the token if within other content. Although maybe we should just check if it exists in the content at all, and if so we remove the entirety of the content. Since it's unlikely to appear anyways, I bet the only time it'd be in mixed content is something like a URL request to GitHub, which we will know those details from the calling function anyway, so we shouldn't lose any valuable data